app/src/main/java/net/squanchy/home/HomeActivity.java

```diff
@@ -312,6 +295,41 @@ private void trackPageSelection(BottomNavigationSection section) {
         analytics.trackPageView(this, section.name());
     }
 
+    public void requestSignIn() {
+        disposeAllSubscriptions();
+        navigator.toSignInForResult(REQUEST_SIGN_IN_MAY_GOD_HAVE_MERCY_OF_OUR_SOULS);
```

Comment:
It's because Firebase doesn't _really_ unregister its callbacks when you tell it to. It does it _eventually_ (I think it's using a handler and `post()`ing thing around), but not quickly enough. When you sign out, all the nodes' data is invalidated and every active subscription that is observing it will go in error because you need to be logged in at least anonymously to be able to read the data. This is very fast, happens pretty much immediately, even before the user is (re)logged in. So we need to do this hack for now, which is, when we go to signin, we temporarily stop observing for data changes, and then we come back we resume observing.